src/lineup/internal/LineUpPanelActions.ts

```diff
@@ -58,39 +65,46 @@ export default class LineUpPanelActions extends EventHandler {
 
   private idType: IDType | null = null;
 
-  private readonly search: SearchBox<ISearchOption> | null;
+  private readonly searchBoxProvider: SearchBoxProvider;
 
   readonly panel: SidePanel | null;
-  readonly node: HTMLElement;
+  readonly node: HTMLElement; // wrapper node
+
+  private readonly header: PanelHeader;
+  private readonly tabContainer: PanelTabContainer;
+
   private overview: HTMLElement;
   private wasCollapsed = false;
 
   constructor(protected readonly provider: LocalDataProvider, ctx: any, private readonly options: Readonly<IARankingViewOptions>, doc = document) {
     super();
 
-    if (options.enableAddingColumns) {
-      this.search = new SearchBox<ISearchOption>({
-        placeholder: i18n.t('tdp:core.lineup.LineupPanelActions.searchPlaceholder')
-      });
-      this.search.on(SearchBox.EVENT_SELECT, (item) => {
-        this.node.querySelector('.lu-adder')!.classList.remove('once');
-        item.action();
-      });
-    }
+    this.node = doc.createElement('aside');
+    this.node.classList.add('lu-side-panel-wrapper');
+
+    this.header = new PanelHeader(this.node);
+
+    this.searchBoxProvider = new SearchBoxProvider();
+
+    if (this.options.enableSidePanel === 'top') {
+      this.node.classList.add('lu-side-panel-top');
 
-    if (this.options.enableSidePanel !== 'top') {
-      this.panel = new SidePanel(ctx, doc, {
-        chooser: false
-      });
-      this.node = this.panel.node;
     } else {
-      this.node = doc.createElement('div');
-      this.node.classList.add('lu-side-panel', 'lu-side-panel-top');
+      const sidePanel = new SidePanelTab(this.node, this.searchBoxProvider.createSearchBox(), ctx, doc);
+      this.panel = sidePanel.panel;
+
+      this.tabContainer = new PanelTabContainer(this.node);
+      this.tabContainer.addTab(sidePanel);
+      this.tabContainer.showTab(sidePanel);
     }
-    this.node.classList.add('tdp-view-lineup');
-    this.collapse = options.enableSidePanel === 'top' || options.enableSidePanel === 'collapsed';
 
     this.init();
+    this.collapse = options.enableSidePanel === 'top' || options.enableSidePanel === 'collapsed';
+  }
+
+  removeTabButtonHighlighting() {
+    this.tabContainer.showDefault();
```

Comment:
It's a bit confusing that the header has to remove the highlight. Especially, because it is directly accessing the CSS classes of the tab. We might have to change the architecture of the tabs a bit. Maybe it helps if the `PanelNavButton` has a reference to his own tab or the `PanelTab` has an (optional) reference to the button that can activate it.